core/src/main/java/org/elasticsearch/search/aggregations/AggregationBuilders.java

```diff
@@ -159,6 +161,22 @@ public static FiltersAggregationBuilder filters(String name, KeyedFilter... filt
     public static FiltersAggregationBuilder filters(String name, QueryBuilder... filters) {
         return new FiltersAggregationBuilder(name, filters);
     }
+    
+    /**
+     * Create a new {@link AdjacencyMatrix} aggregation with the given name.
+     */
+    public static AdjacencyMatrixAggregationBuilder adjacencyMatrix(String name, 
+            org.elasticsearch.search.aggregations.bucket.adjacency.AdjacencyMatrixAggregator.KeyedFilter... filters) {
```

Comment:
Maybe we could even take a `Map<String, QueryBuilder>` to avoid exposing the KeyedFilter class in the client API.